src/mlpack/core/util/deprecated.hpp

```diff
@@ -0,0 +1,19 @@
+/**
+ * @file deprecated.hpp
+ * @author Marcos Pividori.
+ *
+ * Definition of the DEPRECATED macro.
+ */
+#ifndef MLPACK_CORE_UTIL_DEPRECATED_HPP
+#define MLPACK_CORE_UTIL_DEPRECATED_HPP
+
+#ifdef __GNUG__
+#define DEPRECATED __attribute__((deprecated))
+#elif defined(_MSC_VER)
+#define DEPRECATED __declspec(deprecated)
+#else
+#pragma message("WARNING: You need to implement DEPRECATED for this compiler")
+#define DEPRECATED
```

Comment:
Yes, I agree! Done in: https://github.com/mlpack/mlpack/pull/765/commits/84195797da5a8ade52a54891add590d3be7d606e